### ISSUE TITLE
Docker image is to large

### ISSUE THREAD

**Issue by mkinney** [sigp/lighthouse#748]
## Description

If we utilize multi-stage builds, we can shrink the docker image from 1.87GB to something much smaller.

## Present Behaviour

When you run `docker build .` you get a very large docker image.

## Expected Behaviour

Smaller is better.

## Steps to resolve

`docker build .` will build the docker image locally
`docker images` will show size

**Comment by mkinney** [comment]
Research:

``` docker images | grep sigp
sigp/lighthouse                           latest              cff168b65387        4 hours ago         1.87GB
```
Dockerfile is based on `rust:1.39.0`

```$ docker run -it sigp/lighthouse bash
root@ab301c12be23:/# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"```

debian has a buster-slim which is only 69mb

```$ docker images | grep slim
debian                                    buster-slim         2dbffcb4f093        3 weeks ago         69.2MB```

The lighthouse binary is only 37mb:
```# ls -alh /usr/local/cargo/bin/lighthouse
-rwxr-xr-x 1 root root 37M Dec 20 04:40 /usr/local/cargo/bin/lighthouse```

Note: If we switch to using alpine we might be able to shave another 60mb.

```$ docker images alpine
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
alpine              latest              965ea09ff2eb        8 weeks ago         5.55MB```

**Comment by mkinney** [comment]
I should have a PR shortly.

**Comment by mkinney** [comment]
Looks like I can get it down to 138mb using debian:buster-slim.

`Step 6/6 : COPY --from=builder /usr/local/cargo/bin/lighthouse /usr/local/bin/lighthouse
 ---> 659e7cf0d890
Successfully built 659e7cf0d890

mikekinneymbp:lighthouse mikekinney$ docker images
REPOSITORY                                TAG                 IMAGE ID            CREATED             SIZE
<none>                                    <none>              659e7cf0d890        12 seconds ago      138MB`

**Comment by mkinney** [comment]
`$ docker run -it 659e7cf0d890 lighthouse bn --http testnet -r quick 8 `date +%s`
Dec 20 05:07:16.146 WARN Ethereum 2.0 is pre-release. This software is experimental.
Dec 20 05:07:16.522 WARN Block production impaired               reason: dummy eth1 backend is enabled`

**Comment by paulhauner** [comment]
Resolved by #749 

Thank you @mkinney!
